### PR TITLE
Schema

### PR DESCRIPTION
**Pull request by richardmatthewsdev**
Add the ability to define a set of fields in a schema that can be used in a Transformation

### CODE REVIEW

File: app/frontend/js/apps/SchemaApp/SchemaApp.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+// actions from state
+
+import { selectSchemaFieldIds } from '~/js/features/SchemaApp/SchemaFieldsSlice';
+
+const SchemaApp = () => {
+  const schemaFieldIds = useSelector(selectSchemaFieldIds);
+
+  return (
+    <>
```
Comment (danielaboost) [inline app/frontend/js/apps/SchemaApp/SchemaApp.jsx:17]: is this empty tag a redux thing or an accident? 🤔

File: app/frontend/js/apps/SchemaApp/components/SchemaField.jsx [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        schemaFieldId: id,
+        schemaId: appDetails.schema.id
+      })
+    );
+
+    setSchemaFieldValue('');
+  };
+
+  return (
+    <>
```
Comment (danielaboost) [inline app/frontend/js/apps/SchemaApp/components/SchemaField.jsx:119]: Same question as above for this one!

File: app/helpers/schema_fields_helper.rb
Comment (danielaboost) [inline app/helpers/schema_fields_helper.rb:1]: @richardmatthewsdev did this get left behind?

File: app/views/schemas/edit.html.erb
Comment (danielaboost) [inline app/views/schemas/edit.html.erb:1]: Is this also leftovers?

File: app/views/schemas/new.html.erb
Comment (danielaboost) [inline app/views/schemas/new.html.erb:1]: Is this also left over or does it interact with JSX in some way I don't know about 🤔

**Review by danielaboost — APPROVED** [review]

**Comment by danielaboost** [comment]
@richardmatthewsdev approved! Thank you so so much for your hard work - after having a play around with this feature its already making building pipelines faster 😁 

A small suggestion - it would be awesome if the defined values added to a schema field display in alphabetical order in both the list on the schema page, and the select when its on a model (I noticed yesterday with 46 agencies it would make this data a bit easier to work with).

I know you've spent a lot of time on this, so we can do it as a different story! GREAT WORK ✨

File: app/frontend/js/apps/SchemaApp/components/SchemaField.jsx [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+  setActiveSchemaField,
+} from "~/js/features/SchemaApp/UiSchemaFieldsSlice";
+
+
+import { addSchemaFieldValue } from '~/js/features/SchemaApp/SchemaFieldValuesSlice';
+
+import Button from "react-bootstrap/Button";
+import Modal from "react-bootstrap/Modal";
+
+import SchemaFieldValue from "~/js/apps/SchemaApp/components/SchemaFieldValue";
```
Comment (eoin-boost) [inline app/frontend/js/apps/SchemaApp/components/SchemaField.jsx:23]: could a shorter relative reference be used?

File: app/frontend/js/apps/SchemaApp/components/SchemaFieldNavigationPanel.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { selectAllSchemaFields } from "~/js/features/SchemaApp/SchemaFieldsSlice";
+import SchemaFieldNavigationListItem from "./SchemaFieldNavigationListItem";
+import AddSchemaField from "~/js/apps/SchemaApp/components/AddSchemaField";
```
Comment (eoin-boost) [inline app/frontend/js/apps/SchemaApp/components/SchemaFieldNavigationPanel.jsx:5]: relative ref could be good again?

File: app/views/schemas/new.html.erb
Comment (eoin-boost) [inline app/views/schemas/new.html.erb:1]: assuming we can get rid of this file?

**Review by eoin-boost — APPROVED** [review]
great work @richardmatthewsdev

File: app/frontend/js/apps/SchemaApp/SchemaApp.jsx (hunk 1 above)
Comment (richardmatthewsdev) [reply app/frontend/js/apps/SchemaApp/SchemaApp.jsx:17]: This is a way of returning multiple HTML elements in one component without having to wrap them in an empty div :)

File: app/helpers/schema_fields_helper.rb
Comment (richardmatthewsdev) [reply app/helpers/schema_fields_helper.rb:1]: Thanks!

File: app/views/schemas/edit.html.erb
Comment (richardmatthewsdev) [reply app/views/schemas/edit.html.erb:1]: Thanks!

**Pull request merged**